pkg/addonmanager/addonclients/fluxaddonclient.go

```diff
@@ -52,11 +52,31 @@ const (
 	deleteClusterconfigCommitMessage  = "Delete commit of cluster configuration; generated by EKS-A CLI"
 )
 
+type GitClient interface {
+	Add(filename string) error
+	Remove(filename string) error
+	Clone(ctx context.Context) error
+	Commit(message string) error
+	Push(ctx context.Context) error
```

Comment:
Thought you mentioned `GitClient` is for local git operations while `GitProviderClient` for remote ops.
Doesn't Push and Pull interact with remote repo? Or is `GitClient` interface for go-git client while `GitProviderClient` for operations go-git does not handle?

Reply:
yes, that's right.

GitClient is for operations that are provider-agnostic (like push/pull/init/clone/etc) while GitProviderClient is for provider dependent operations (like create repo)

Reply:
If it's the latter, wonder if we can name  `GitClient` differently like `GoGitClient`

Reply:
well I want it to be a generic client for any provider-agnostic Git actions, akin to the git CLI executable, so I think calling it just `GitClient` makes sense; the idea is to keep GoGit behind the scenes as an implementation detail

Reply:
Yea make sense, do u mind adding doc explaining what each interface is responsible for?